cfpb/consumerfinance.gov
# Fix functional tests on staging

This PR makes a couple of separate changes intended to fix our functional test runs on staging:

1. Fixes the data-option lookup for categories in the filterable list blog tests. This change is to how we look up the `data-option` from human-friend strings by converting the human-friendly string to kebab-case. 
   
   The issue was, “Data, Research, and Policy” was getting converted to `data,-research,-and-policy`, when the actual `data-option` value is `data-research-policy`. 

   The fix here is wrapped in a utility function to perform this formatting by using a regex to drop words (“and”) and characters (“,”, “’”) before kebab-casing the resulting string.
   
   I’ve made use of this function in all the places where we lookup categories and topics from human-friendly strings.

2. I've added [`@cypress/skip-test`](https://github.com/cypress-io/cypress-skip-test) to let us conditionally skip tests in different environments, and applied that to the FIG tests. Currently they'll skip when running on staging.

@wpears @contolini Can you confirm I've added the `skip-test` package correctly? @contolini do you have any concerns with how I've skipped the FIG tests on staging? 

I've also added the `CYPRESS_ENVIRONMENT=github` to our functional test GHA workflow so that we could potentially transition to `skipOn('github')` in place of our big list of specs, but I have not made that change here.

## How to test this PR

To run the tests against staging with the `CYPRESS_ENVIRONMENT`: 

```
CYPRESS_ENVIRONMENT=staging yarn cypress run --config baseUrl=https://[STAGING_URL]/
```

All the tests should now pass.

## Checklist
